code generators/de.fhdo.lemma.model_processing.code_generation.springcloud.eureka/src/main/java/de/fhdo/lemma/model_processing/code_generation/springcloud/eureka/code_generation/EurekaCodeGenerator.xtend

```diff
@@ -0,0 +1,191 @@
+
+        /*
+         * Create a PropertyFile based on the intermediate operation model configuration and merge
+         * it with the potentially existing one.
+         */
+        val serviceConfig = service.node.getEffectiveConfigurationValues(service)
+        val eurekaConfig = EurekaTemplate::getPropertiesForEurekaServiceConfig(serviceConfig)
+        val properties = new SortableProperties
+        properties.load(new ByteArrayInputStream(eurekaConfig.getBytes()))
+        val propertyFile = new PropertyFile(path.toString, properties)
+        val mergedFile = OpenedPropertyFiles.instance.mergePropertyFile(propertyFile)
+        OpenedPropertyFiles.instance.add(mergedFile)
+    }
+
+    /**
+     * Create an IntermediateInfrastructure-specific Kubernetes deployment file.
+     */
+    private def createEurekaKubernetesDeployment(IntermediateInfrastructureNode node) {
+        val kubernetes = EurekaTemplate::getKubernetesDeploymentForEureka(node)
+        val path = '''«targetFolder»«File.separator»«node.name»«File.separator»''' +
+            '''«node.name.toLowerCase»-deployment.yaml'''
+
+        content.put(path, kubernetes)
+    }
+
+    /**
+     * Create an IntermediateOperationNode-specific Docker-Compose deployment file.
+     */
+    private def createDockerComposeDeployment(IntermediateInfrastructureNode node) {
+        /*
+         * Set targetFolder for Docker-Compose file and open an existing or create a new
+         * Docker-Compose file
+         */
+        OpenedDockerComposeFile.instance.init(targetFolder)
+        OpenedDockerComposeFile.instance.openExistingDockerComposeFile
+
+        // Build Docker-Compose part based on a operation aspect or node configuration
+        val dockerComposeAspect = node.aspects.findFirst[aspect | aspect.name == "ComposePart"]
+        if (dockerComposeAspect === null)
+            OpenedDockerComposeFile.instance.addOrReplaceDockerComposePart(node.name,
+                EurekaTemplate::getDockerComposeForEureka(node))
+        else
+            OpenedDockerComposeFile.instance.addOrReplaceDockerComposePart(node.name,
+                dockerComposeAspect.propertyValues?.get(0).value)
+
+        // Write Docker-Compose file to the file path
+        val filePath = '''«targetFolder»«File.separator»docker-compose.yml'''
+        content.put(filePath, OpenedDockerComposeFile.instance.toString)
+    }
+
+    /**
+     * Generate Eureka infrastructure node based on the spring cloud framework implemented in
+     * Kotlin.
+     */
+    private def createEurekaComponents(IntermediateInfrastructureNode node) {
+        // Generate pom.xml file
+        val pomFilePath = '''«targetFolder»«File.separator»«node.name.toLowerCase»''' +
+            '''«File.separator»pom.xml'''
+        this.content.put(pomFilePath, EurekaTemplate.generatePOMFile(node))
+
+        // generate main class for the Spring-based Eureka implementation in Kotlin
```

Comment:
Again as for the Zuul generator: Please open an issue to also enable the generation of Java-based Eureka services in the future.

Reply:
Added the option to also generate a Java-based Eureka implementation.